File: register_tool.cpp
#include <iostream>
#include <string>

using namespace std;

int main() {
    string name, email, password, roleChoice, role, adminKey;
    string secret = "MRS_ADMIN_2026";

    cout << "--- MRS Registration Payload Generator (C++) ---" << endl;
    
    cout << "Enter Name: ";
    getline(cin, name);
    
    cout << "Enter Email: ";
    getline(cin, email);
    
    cout << "Enter Password: ";
    getline(cin, password);
    
    cout << "Select Role (1: Student, 2: Teacher, 3: Admin): ";
    getline(cin, roleChoice);

    if (roleChoice == "3") {
        role = "Admin";
        adminKey = secret;
    } else if (roleChoice == "2") {
        role = "Teacher";
    } else {
        role = "Student";
    }

    // Construct JSON manually
    string json = "{";
    json += "\"name\": \"" + name + "\",";
    json += "\"email\": \"" + email + "\",";
    json += "\"password\": \"" + password + "\",";
    json += "\"role\": \"" + role + "\"";
    if (role == "Admin") {
        json += ",\"adminKey\": \"" + adminKey + "\"";
    }
    json += "}";

    cout << "\nGenerated JSON Payload:\n" << json << endl;
    cout << "\nTo execute with CURL:\ncurl -X POST http://localhost:8000/api/register -H \"Content-Type: application/json\" -d '" << json << "'" << endl;

    return 0;
}

File: register_tool.py
import requests
import json

# Configuration
API_URL = "http://localhost:8000/api/register"
ADMIN_SECRET = "MRS_ADMIN_2026"

def register_user():
    print("--- MRS Registration Tool (Python) ---")
    name = input("Enter Name: ")
    email = input("Enter Email: ")
    password = input("Enter Password: ")
    
    print("\nSelect Role:")
    print("1. Student")
    print("2. Teacher")
    print("3. Admin")
    role_choice = input("Choice (1-3): ")
    
    role_map = {"1": "Student", "2": "Teacher", "3": "Admin"}
    role = role_map.get(role_choice, "Student")
    
    payload = {"name": name, "email": email, "password": password, "role": role}
    
    if role == "Admin":
        payload["adminKey"] = ADMIN_SECRET
        print(f"Auto-filling Admin Key: {ADMIN_SECRET}")

    try:
        response = requests.post(API_URL, json=payload)
        print(f"\nResponse [{response.status_code}]: {response.text}")
    except Exception as e:
        print(f"Error: {e}")

if __name__ == "__main__":
    register_user()